admin/weaviate/docker-compose.yml

```diff
@@ -0,0 +1,59 @@
+# Docker Compose for Weaviate with RBAC enabled
+# This configuration sets up Weaviate with authentication and authorization
+# for the Edutelligence microservices
+
+version: '3.8'
+
+services:
+  weaviate:
+    image: semitechnologies/weaviate:1.22.1
+    container_name: edutelligence-weaviate
+    restart: on-failure:0
+    ports:
+      - "8080:8080"
+      - "50051:50051"
+    environment:
+      # Query and clustering configuration
+      QUERY_DEFAULTS_LIMIT: 25
+      QUERY_MAXIMUM_RESULTS: 10000
+      AUTHENTICATION_ANONYMOUS_ACCESS_ENABLED: 'false'
+      PERSISTENCE_DATA_PATH: '/var/lib/weaviate'
+      DEFAULT_VECTORIZER_MODULE: 'none'
+      ENABLE_MODULES: 'backup-filesystem,text2vec-cohere,text2vec-huggingface,text2vec-palm,text2vec-openai,generative-openai,generative-cohere,generative-palm,ref2vec-centroid,reranker-cohere,qna-openai'
+      CLUSTER_HOSTNAME: 'node1'
+      
+      # Authentication configuration  
+      AUTHENTICATION_APIKEY_ENABLED: 'true'
+      AUTHENTICATION_APIKEY_ALLOWED_KEYS: 'admin-key'
+      AUTHENTICATION_APIKEY_USERS: 'admin@edutelligence.local'
+      
+      # Authorization/RBAC configuration
+      AUTHORIZATION_ADMINLIST_ENABLED: 'true'
+      AUTHORIZATION_ADMINLIST_USERS: 'admin@edutelligence.local'
+      AUTHORIZATION_ADMINLIST_READONLY_USERS: ''
```

Comment:
Why are you using adminlist here? According to the documentation it is mutually exclusive with RBAC authorization. For RBAC you need to use `AUTHORIZATION_RBAC_ENABLED` and `AUTHORIZATION_RBAC_ROOT_USERS `.